proposals/NNNN-ServiceStatusUpdateToHMI.md

```diff
@@ -0,0 +1,114 @@
+# Service Status Update to HMI
+
+* Proposal: [SDL-NNNN](NNNN-ServiceUpdateToHMI.md)
+* Author: [Robin Kurian](https://github.com/robinmk)
+* Status: **Awaiting Review**
+* Impacted Platforms: [Core / RPC]
+
+## Introduction
+
+This proposal is about adding a new RPC called 'OnServiceUpdate' which will be used by the HMI layer to know the status of certain services from SDL Core.
+
+
+## Motivation
+
+Today, when a Mobile Navigation application is activated there is a series of steps executed begining with the app sending the StartService request for the Video Service and ending with the module acknowledging this request in a successful scenario. These steps include getting the current system time, performing a policy table update, decrypting certificates and ensuring validity of the certificates. Any of these steps could fail and since there is no mechanism for updating HMI with the status of the service, the HMI has no means of providing accurate feedback to the user on the status of the system or what steps to take in case of an error.
+The motivation behind this proposal is to over come this issue by introducing a new notification between SDL and HMI.
+
+
+## Proposed solution
+
+
+### Additions to HMI_API
+
+The proposed solution is to introduce a new RPC for SDL Core to update HMI with the status of a particular service.
+
+```xml
+<function name="OnServiceUpdate" messagetype="notification">
+  <description>
+    Must be sent by SDL to HMI when there is an update on status of certain services.
+    Services supported with current version: Video
+  </description>
+  <param name="serviceType" type="Common.ServiceType" mandatory="true">
+    <description>Specifies the service which has been updated.</description>
+  </param>
+  <param name="serviceEvent" type="Common.ServiceEvent" mandatory="false">
+    <description>Specifies service update event.</description>
+  </param>
+  <param name="reason" type="Common.ServiceUpdateReason" mandatory="false">
+    <description>
+      The reason for a service event. Certain events may not have a reason such as when a service is ACCEPTED (which is the normal expected behavior).
```

Comment:
Change to:
The reason for a service event. Certain events may not have a reason, such as when a service is ACCEPTED (which is the normal expected behavior).